src/LabelTrack.cpp

```diff
@@ -422,85 +428,171 @@ ImportExportPrefs::RegisteredControls reg{ wxT("LabelStyle"), AddControls };
 
-      title = token;
-   }
+         sr.setTimes( t0, t1 );
+
+         title = token;
+      }
+
+      // Newer selection fields are written on additional lines beginning with
+      // '\' which is an impossible numerical character that older versions of
+      // audacity will ignore.  Test for the presence of such a line and then
+      // parse it if we can.
 
-   // Newer selection fields are written on additional lines beginning with
-   // '\' which is an impossible numerical character that older versions of
-   // audacity will ignore.  Test for the presence of such a line and then
-   // parse it if we can.
-
-   // There may also be additional continuation lines from future formats that
-   // we ignore.
-
-   // Advance index over all continuation lines first, before we might throw
-   // any exceptions.
-   int index2 = index;
-   while (index < (int)file.GetLineCount() &&
-          file.GetLine(index).StartsWith(continuation))
-      ++index;
-
-   if (index2 < index) {
-      wxStringTokenizer toker { file.GetLine(index2++), wxT("\t") };
-      auto token = toker.GetNextToken();
-      if (token != continuation)
+      // There may also be additional continuation lines from future formats that
+      // we ignore.
+
+      // Advance index over all continuation lines first, before we might throw
+      // any exceptions.
+      int index2 = index;
+      while (index < (int)file.GetLineCount() &&
+             file.GetLine(index).StartsWith(continuation))
+         ++index;
+
+      if (index2 < index) {
+         wxStringTokenizer toker { file.GetLine(index2++), wxT("\t") };
+         auto token = toker.GetNextToken();
+         if (token != continuation)
+            throw BadFormatException{};
+
+         token = toker.GetNextToken();
+         double f0;
+         if (!Internat::CompatibleToDouble(token, &f0))
+            throw BadFormatException{};
+
+         token = toker.GetNextToken();
+         double f1;
+         if (!Internat::CompatibleToDouble(token, &f1))
+            throw BadFormatException{};
+
+         sr.setFrequencies(f0, f1);
+      }
+      break;
+   }
+#ifdef EXPERIMENTAL_SUBRIP_LABEL_FORMATS
+   case LabelFormat::SUBRIP:
+   {
+      if ((int)file.GetLineCount() < index + 2)
          throw BadFormatException{};
 
-      token = toker.GetNextToken();
-      double f0;
-      if (!Internat::CompatibleToDouble(token, &f0))
+      long identifier;
+      // The first line should be a numeric counter; we can ignore it otherwise
+      if (!firstLine.ToLong(&identifier))
          throw BadFormatException{};
 
-      token = toker.GetNextToken();
-      double f1;
-      if (!Internat::CompatibleToDouble(token, &f1))
+      wxString timestamp = file.GetLine(index++);
+      // Parsing data of the form 'HH:MM:SS,sss --> HH:MM:SS,sss'
+      // Assume that the line is in exactly that format, with no extra whitespace
+      // and less than 24 hours.
+      if (timestamp.length() != 29)
```

Comment:
While I have never seen a vtt or srt file do anything other than this, I'm a bit uncomfortable with such an assumption. I'd prefer something more robust here.